src/audio/drc/drc.c

```diff
@@ -178,23 +176,26 @@ static struct comp_dev *drc_new(const struct comp_driver *drv,
 	cd->model_handler = comp_data_blob_handler_new(dev);
 	if (!cd->model_handler) {
 		comp_cl_err(&comp_drc, "drc_new(): comp_data_blob_handler_new() failed.");
-		rfree(dev);
-		rfree(cd);
-		return NULL;
+		goto cd_fail;
```

Comment:
why do you need to go to cd_fail in this case? cd->model_handler was never successfully alloc'ed anyway right?

Reply:
`cd` still needs to be freed; `comp_data_blob_handler_free(NULL)` is no-op.
(My mental model for `cd_fail` was: jump here if cd is allocated)

Or we can move `rfree(cd)` below `fail:` instead.

Reply:
@ranj063 `free(NULL)` handlers are no-ops and since `cd was allocated with `rzalloc` it is `NULL`. So this is just a matter of flavour. Label could be above `cd` but would serve same purpose.